NorthstarDLL/mods/autodownload/moddownloader.h

```diff
@@ -0,0 +1,141 @@
+	char* modsListUrl;
+
+	struct VerifiedModVersion
+	{
+		std::string checksum;
+	};
+	struct VerifiedModDetails
+	{
+		std::string dependencyPrefix;
+		std::unordered_map<std::string, VerifiedModVersion> versions = {};
+	};
+	std::unordered_map<std::string, VerifiedModDetails> verifiedMods = {};
+
+	/**
+	 * Downloads a mod archive from distant store.
+	 *
+	 * This rebuilds the URI of the mod archive using both a predefined store URI
+	 * and the mod dependency string from the `verifiedMods` variable, or using
+	 * input mod name as mod dependency string if bypass flag is set up; fetched
+	 * archive is then stored in a temporary location.
+	 *
+	 * If something went wrong during archive download, this will return an empty
+	 * optional object.
+	 *
+	 * @param modName name of the mod to be downloaded
+	 * @param modVersion version of the mod to be downloaded
+	 * @returns location of the downloaded archive
+	 */
+	std::optional<fs::path> FetchModFromDistantStore(std::string_view modName, std::string_view modVersion);
+
+	/**
+	 * Checks whether a mod is verified.
+	 *
+	 * A mod is deemed verified/authorized through a manual validation process that is
+	 * described here: https://github.com/R2Northstar/VerifiedMods; in practice, a mod
+	 * is considered authorized if their name AND exact version appear in the
+	 * `verifiedMods` variable.
+	 *
+	 * @param modName name of the mod to be checked
+	 * @param modVersion version of the mod to be checked, must follow semantic versioning
+	 * @returns whether the mod is authorized and can be auto-downloaded
+	 */
+	bool IsModAuthorized(std::string_view modName, std::string_view modVersion);
+
+	/**
+	 * Tells if a mod archive has not been corrupted.
+	 *
+	 * The mod validation procedure includes computing the SHA256 hash of the final
+	 * archive, which is stored in the verified mods list. This hash is used by this
+	 * very method to ensure the archive downloaded from the Internet is the exact
+	 * same that has been manually verified.
+	 *
+	 * @param modPath path of the archive to check
+	 * @param expectedChecksum checksum the archive should have
+	 * @returns whether archive is legit
+	 */
+	bool IsModLegit(fs::path modPath, std::string_view expectedChecksum);
+
+	/**
+	 * Extracts a mod archive to the game folder.
+	 *
+	 * This extracts a downloaded mod archive from its original location to the
+	 * current game profile, in the remote mods folder.
+	 *
+	 * @param modPath location of the downloaded archive
+	 * @returns nothing
+	 */
+	void ExtractMod(fs::path modPath);
+
+  public:
+	ModDownloader();
+
+	/**
+	 * Retrieves the verified mods list from the central authority.
+	 *
+	 * The Northstar auto-downloading feature does NOT allow automatically installing
+	 * all mods for various (notably security) reasons; mods that are candidate to
+	 * auto-downloading are rather listed on a GitHub repository
+	 * (https://raw.githubusercontent.com/R2Northstar/VerifiedMods/master/mods.json),
```

Comment:
Maybe refer to the const instead of copy/pasting the link but tbh this is just pure bike-shedding and not blocking.